### PR TITLE
Devolved registers of interests

### PR DESCRIPTION
**Pull request by ajparsons**
This PR adds new scrapers and a common data format for registers of interests across the devolved Parliaments (+ London partially).

The goal is to import registers of interests as json files with a more complex structure to move formatting into the TWFY template.

Here we add use [a data structure](https://github.com/mysociety/mysoc-validator/blob/main/src/mysoc_validator/models/interests.py#L426) that all the different scrapers work with, can dump to json - and this will then be stored in the twfy database.

All scrapers create this json as well as the equivalent XML as this will continue to be how the comparison over time is produced. There is a one-off conversion script for old XML to new-style json for consistency on display of old MP data. 

The readme.MD has more information on the different scrapers (most of which use APIs). London is currently incomplete because we can't guarantee the equivalent TWFY IDs. Included at this point to test the flexibility of the format. 

To create the initial data:

```
poetry run python -m pyscraper.regmem download-all-registers --chamber commons
poetry run python -m pyscraper.regmem download-all-registers --chamber scotland --force-refresh
poetry run python -m pyscraper.regmem download-all-registers --chamber ni
poetry run python -m pyscraper.regmem download-latest-register --chamber senedd
poetry run python -m pyscraper.regmem convert-legacy-xml-to-json
```

### CODE REVIEW

File: scripts/dailyupdate [hunk 1]
```diff
@@ -7,6 +7,10 @@ source ~/parlparse/scripts/consts
 # Update register of members interests
 cd ~/parlparse
 poetry run python -m pyscraper.regmem download-all-registers --chamber commons --quiet
+poetry run python -m pyscraper.regmem download-latest-register --chamber scotland --quiet
```
Comment (struan) [inline scripts/dailyupdate:10]: why is the commons fetching all registers and the rest only fetching the latest?

File: pyscraper/regmem/__main__.py [hunk 2]
```diff
@@ -21,10 +24,22 @@ def download_all_registers(
     quiet: bool = False,
     no_progress: bool = False,
 ):
-    if chamber == "commons":
+    """
+    Create all registers for a chamber - including fetching or trying to
+    derive historical data where possible.
+    """
+    if chamber in "commons":
```
[the remark this answers is not in the document]
Comment (ajparsons) [reply pyscraper/regmem/__main__.py:31]: Yep, wrong - although works like this surprisingly.

File: scripts/dailyupdate (hunk 1 above)
Comment (ajparsons) [reply scripts/dailyupdate:10]: No strong reason - as it runs daily they should all download-latest-register

File: pyscraper/regmem/commons/process.py [hunk 3]
```diff
@@ -289,91 +393,82 @@ def get_restacked_register(self) -> list[Member]:
...
+        return uni
 
-    def write_mysoc_regmem(self, force_refresh: bool = False, quiet: bool = False):
+    def write_universal_regmem(self, force_refresh: bool = False, quiet: bool = False):
         """
         Write the register to the parldata directory
         """
 
-        dest_folder = self.parldata_dir / "scrapedxml" / "regmem"
+        dest_folder = self.parldata_dir / "scrapedjson" / "common_regmem" / "commons"
```
[the remark this answers is not in the document]
Comment (ajparsons) [reply pyscraper/regmem/commons/process.py:458]: I changed it from universal to common as I went - will change universal refs given other comments - don't think it's in other files.

File: pyscraper/regmem/__main__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dracos) [reply pyscraper/regmem/__main__.py:31]: (`in` on strings is a substring test; note the edge case that "" would be True)

File: pyscraper/regmem/commons/process.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajparsons) [reply pyscraper/regmem/commons/process.py:458]: but maybe common_regmem is wrong given one of the items is 'commons' - maybe 'json_regmem'

File: pyscraper/regmem/commons/process.py [hunk 4]
```diff
@@ -3,45 +3,61 @@
...
+    """
+    if not name:
+        return ""
+    new_text = ""
+    for char in name:
+        if char.isupper():
+            new_text += " "
+        new_text += char
+    nice_name = new_text.strip()
+    return nice_name
```
[the remark this answers is not in the document]
Comment (ajparsons) [reply pyscraper/regmem/commons/process.py:53]: Super - added.

**Comment by ajparsons** [comment]
Ok! All of those should have been addressed and merged back in.

Points that will need to be adjusted in the twfy PR - changing 'common_regmem' to 'universal_format_regmem' and the commons jsons will now be in a 'house-of-commons' folder rather than just commons.
